src/ctia/stores/es/crud.clj

```diff
@@ -92,23 +87,43 @@
                     (build-create-result model coerce-fn)))
                 (remove-es-actions items) models)}))
 
-
-(s/defn get-docs-with-indices
+(defn get-docs-with-indices
   "Retrieves a documents from a search \"ids\" query. It enables to retrieves
  documents from an alias that points to multiple indices.
 It returns the documents with full hits meta data including the real index in which is stored the document."
-  [{:keys [conn index]} :- ESConnState
-   ids :- [s/Str]
+  [{:keys [conn index] :as _conn-state}
+   ids
    es-params]
   (let [ids-query (q/ids (map ensure-document-id ids))
         res (ductile.doc/query conn
                                index
                                ids-query
                                (assoc (make-es-read-params es-params)
+                                      :limit (count ids)
                                       :full-hits?
                                       true))]
     (:data res)))
 
+(defn lazy-get-docs-with-indices
+  "Retrieves a documents from a search \"ids\" query. It enables to retrieves
+ documents from an alias that points to multiple indices.
+It returns the documents with full hits meta data including the real index in which is stored the document."
```

Comment:
This docstring sounds like a hip-hop song :)

Reply:
I do live in the wood, but I grew up in the hood